xapian-core/weight/tfidfweight.cc

```diff
@@ -70,26 +74,48 @@ TfIdfWeight::name() const
 string
 TfIdfWeight::serialise() const
 {
-    return normalizations;
+    if (param_s > 0 || param_delta > 0) {
+	string result = serialise_double(param_s);
+	result += serialise_double(param_delta);
+	return result;
+    } else {
+	return normalizations;
+    }
 }
 
 TfIdfWeight *
-TfIdfWeight::unserialise(const string & s) const
+TfIdfWeight::unserialise(const string & str) const
```

Comment:
`slope` would be better; single-character variable names should generally be avoided. (`i`, `j` for indices is okay, but for everything else being more explicit is generally more readable.)